provider/bikeshed/src/units/essentials.rs

```diff
@@ -18,6 +19,54 @@ use zerovec::ZeroMap;
 
 impl DataProvider<UnitsEssentialsV1Marker> for DatagenProvider {
     fn load(&self, req: DataRequest) -> Result<DataResponse<UnitsEssentialsV1Marker>, DataError> {
+        /// Fills the prefixes map with the SI prefixes (binary and decimal)
+        ///
+        /// # Errors
+        ///
+        /// - `DataError::custom("Invalid prefix: must be Binary or Decimal")` if the prefix is not Binary or Decimal
+        /// - `DataError::custom("Failed to parse pattern key value")` if the pattern key value cannot be parsed
+        fn fill_si(
+            data: &BTreeMap<String, Patterns>,
+            prefixes_map: &mut BTreeMap<PatternKey, String>,
+        ) -> Result<(), DataError> {
+            const BINARY_PREFIX: &str = "1024p";
+            const DECIMAL_PREFIX: &str = "10p";
+
+            for (key, patterns) in data.iter().filter(|(key, _)| {
+                key.starts_with(BINARY_PREFIX) || key.starts_with(DECIMAL_PREFIX)
+            }) {
+                let pattern_key = if let Some(trimmed_key) = key.strip_prefix(BINARY_PREFIX) {
+                    trimmed_key
+                        .parse::<u8>()
+                        .map(PatternKey::Binary)
+                        .map_err(|_| {
+                            DataError::custom("Failed to parse pattern key value")
+                                .with_debug_context(&key)
+                        })?
```

Comment:
you can move this out of the if-else, so you don't need it twice

Reply:
done